lib/rules/no-deprecated.js

```diff
@@ -76,21 +76,27 @@ module.exports = {
       deprecated[`${pragma}.PropTypes`] = ['15.5.0', 'the npm module prop-types'];
       // 15.6.0
       deprecated[`${pragma}.DOM`] = ['15.6.0', 'the npm module react-dom-factories'];
-      // 16.3.0
+      // 16.999.0
+      // For now the following life-cycle methods are just legacy, not deprecated:
+      // `componentWillMount`, `componentWillReceiveProps`, `componentWillUpdate`
+      // https://github.com/yannickcr/eslint-plugin-react/pull/1750#issuecomment-425975934
       deprecated.componentWillMount = [
-        '16.3.0',
+        '16.999.0',
```

Comment:
It has to be semver-MINOR, otherwise it doesn't suppress deprecation warnings for the existing React versions (16.3.0 - 16.6.3) where these component lifecycle methods aren't deprecated yet.

Reply:
why? a minor is for adding new options - this isn't adding a new feature, it's just removing warnings for 16.3-16.6 users. Can you explain more?

Reply:
I might have misunderstood your first comment. What semver did you mean? `React` or `eslint-plugin-react`? I agree that current PR can be considered as a patch of `eslint-plugin-react`.

Reply:
I'm talking about this repo :-)

Reply:
Should it be replaced with `16.9.0` now?
https://github.com/facebook/react/blob/master/CHANGELOG.md#react-dom